utils.py

```diff
@@ -48,6 +49,30 @@
 SECONDS_IN_HOUR = 60 * 60
 SECONDS_IN_MINUTE = 60
 
+# During type checking, mypy assumes the variable 'MYPY' to be true.
+# Typing is not a module in python2, so it will only be imported only while type
+# checking.
+
+try:
+    from typing import (
+        Any, Callable, Dict, Generator, Iterable, Iterator, List,
+        Text, Tuple, TypeVar, Union) # isort:skip # pylint: disable=unused-import,import-only-modules
+    T = TypeVar('T')
+    U = TypeVar('U')
+    DICT_LIST_TYPE = TypeVar('DICT_LIST_TYPE', Dict[Any, Any], List[Any])
+except:
+    pass
```

Comment:
@vojtechjelinek Do you have any suggestion how we can handle this?
**Problem**: `typing` module has to be used to add type annotations and it only exists for python3. With this kind of implementation:
1. under normal circumstances (i.e while using python2), the code goes into **except** block and we avoid importing from `typing`. But lint checks and coverage tests fails - lint fails as it runs using python2 and cannot identify `typing`, coverage fails since code inside try block will never execute.
2. when mypy tests are running (currently via github workflow), it uses python3, so the execution goes into **try** block and it is able to perform type checks comfortably.

I don't see a straightforward way of dealing with the problems in (1), do see some way of dealing with this?

Reply:
**Update**: Found a reason behind this - We already have typing module in requirements.txt file so it can work with python2. The issue here is that `install_backend_python_libs` imports `utils.py`, and inside `utils.py` we are importing `typing` module. So when we are installing third part libs in github workflows, `utils.py` is imported and though typing module has not yet been installed but `utils.py` is still trying to import it, which is raising the error.

Reply:
`typing` is a standard library in python3. So I would suggest that we should type annotate `utils.py` only after py3 migration so that we can import `typing` directly. Then we won't face the issue of importing `typing`(which Sajal explained above). I suggest that we must only type annotate those files which are not imported before our third-party libraries are installed.
@vojtechjelinek @seanlip What do you think? Thanks!

Reply:
@hardikkat24 Can you please update the PR description stating the steps you had to take to resolve this error - it is one of the important decisions you took in this PR, and all such important changes should be highlighted via description. (Also leave a comment here after you have edited the PR description so that we know about the update on this issue) 
Thanks!